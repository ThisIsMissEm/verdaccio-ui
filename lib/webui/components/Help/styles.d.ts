/**
 * @prettier
 */
export declare const CardStyled: import("create-emotion-styled").StyledOtherComponent<{}, import("@material-ui/core/Card").CardProps, any>;
export declare const HelpTitle: import("create-emotion-styled").StyledOtherComponent<{}, import("@material-ui/core/Typography").TypographyProps, any>;
