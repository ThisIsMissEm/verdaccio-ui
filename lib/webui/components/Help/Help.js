"use strict";
/**
 * @prettier
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Button_1 = __importDefault(require("@material-ui/core/Button"));
const CardActions_1 = __importDefault(require("@material-ui/core/CardActions"));
const CardContent_1 = __importDefault(require("@material-ui/core/CardContent"));
const Typography_1 = __importDefault(require("@material-ui/core/Typography"));
const react_1 = __importDefault(require("react"));
const url_1 = require("../../utils/url");
const CopyToClipBoard_1 = __importDefault(require("../CopyToClipBoard"));
const styles_1 = require("./styles");
function renderHeadingClipboardSegments(title, text) {
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(Typography_1.default, { variant: 'body2' }, title),
        react_1.default.createElement(CopyToClipBoard_1.default, { text: text })));
}
const Help = () => {
    const registryUrl = url_1.getRegistryURL();
    return (react_1.default.createElement(styles_1.CardStyled, { id: "help-card" },
        react_1.default.createElement(CardContent_1.default, null,
            react_1.default.createElement(Typography_1.default, { component: "h2", gutterBottom: true, id: "help-card__title", variant: "headline" }, "No Package Published Yet."),
            react_1.default.createElement(styles_1.HelpTitle, { color: "textSecondary", gutterBottom: true }, "To publish your first package just:"),
            renderHeadingClipboardSegments('1. Login', `npm adduser --registry ${registryUrl}`),
            renderHeadingClipboardSegments('2. Publish', `npm publish --registry ${registryUrl}`),
            react_1.default.createElement(Typography_1.default, { variant: "body2" }, "3. Refresh this page.")),
        react_1.default.createElement(CardActions_1.default, null,
            react_1.default.createElement(Button_1.default, { color: "primary", href: "https://verdaccio.org/docs/en/installation", size: "small", target: "_blank" }, "Learn More"))));
};
exports.default = Help;
