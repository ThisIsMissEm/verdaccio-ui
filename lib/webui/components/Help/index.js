"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Help_1 = require("./Help");
exports.default = Help_1.default;
