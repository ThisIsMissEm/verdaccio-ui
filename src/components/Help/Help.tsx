/**
 * @prettier
 */

import React from 'react';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import styled from '@emotion/styled';
import Card from '@material-ui/core/Card';

import CopyToClipBoard from '../CopyToClipBoard/CopyToClipBoard';
import { getRegistryURL } from '../../utils/url';

export const CardStyled = styled(Card)`
  && {
    width: 600px;
    margin: auto;
  }
`;

const Help: React.FC = () => {
  const registryUrl = getRegistryURL();

  const renderHeadingClipboardSegments = (title: string, text: string) => (
    <>
      <Typography variant={'body2'}>{title}</Typography>
      <CopyToClipBoard text={text} />
    </>
  );

  return (
    <Card id={'help-card'}>
      <CardContent>
        <Typography component={'h2'} gutterBottom={true} id={'help-card__title'} variant={'headline'}>
          {'No Package Published Yet.'}
        </Typography>
        <Typography color="primary" style={{ marginBottom: '20px' }}>
          {'To publish your first package just:'}
        </Typography>
        {renderHeadingClipboardSegments('1. Login', `$ npm adduser --registry ${registryUrl}`)}
        {renderHeadingClipboardSegments('2. Publish', `$ npm publish --registry ${registryUrl}`)}
        <Typography variant={'body2'}>{'3. Refresh this page.'}</Typography>
      </CardContent>
      <CardActions>
        <Button color={'primary'} href={'https://verdaccio.org/docs/en/installation'} size={'small'} target={'_blank'}>
          {'Learn More'}
        </Button>
      </CardActions>
    </Card>
  );
};

export default Help;
