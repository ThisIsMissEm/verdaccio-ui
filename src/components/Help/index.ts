/**
 * @prettier
 */

export { default } from './Help';
