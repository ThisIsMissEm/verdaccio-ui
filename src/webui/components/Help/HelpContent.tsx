/**
 * @prettier
 */

import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';
import React from 'react';

import CopyToClipBoard from '../CopyToClipBoard'
import OrderedList from '../OrderedList'

interface Props {
  helpSteps: Array<HelpStep>;
}

export interface HelpStep {
  title: string
  command?: string
}

const useStyles = makeStyles({
  item: {
   width: '100%'
  },
});

const HelpContent: React.FC<Props> = ({ helpSteps }) => {
  const classes = useStyles()
  return (
    <OrderedList>
      {helpSteps.map(({ title, command }) => (
        <div className={classes.item} key={title}>
          <Typography variant='body2'>{title}</Typography>
          {command && (
            <CopyToClipBoard text={command} />
          )}
        </div>
      ))}
    </OrderedList>
   )
}

export default HelpContent;
