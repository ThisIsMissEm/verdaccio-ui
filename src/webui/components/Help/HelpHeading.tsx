/**
 * @prettier
 */

import Typography from '@material-ui/core/Typography';
import React from 'react';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  subheading: {
    marginBottom: 20,
  },
});

const HelpSegmentLogin: React.FC = () => {
  const classes = useStyles();
  return (
    <>
      <Typography component="h2" variant="headline" gutterBottom>
        No Package Published Yet.
      </Typography>
      <Typography color="textSecondary" variant="subheading" gutterBottom className={classes.subheading}>
        To publish your first package just:
      </Typography>
    </>
  );
};

export default HelpSegmentLogin;
