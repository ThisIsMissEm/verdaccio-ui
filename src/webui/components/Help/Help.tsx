/**
 * @prettier
 */

import Button from '@material-ui/core/Button';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Card from '@material-ui/core/Card';
import { makeStyles } from '@material-ui/styles';
import React from 'react';

import { getRegistryURL } from '../../utils/url';

import HelpHeading from './HelpHeading';
import HelpContent, { HelpStep } from './HelpContent';

const useStyles = makeStyles({
  card: {
    width: 600,
    margin: 'auto',
  },
});

const Help: React.FC = () => {
  const classes = useStyles();
  const registryUrl = getRegistryURL();
  const publishPackageSteps: Array<HelpStep> = [
    {
      title: 'Login',
      command: `npm adduser --registry ${registryUrl}`,
    },
    {
      title: 'Publish',
      command: `npm publish --registry ${registryUrl}`,
    },
    {
      title: 'Refresh this page',
      command: undefined,
    },
  ];

  return (
    <Card className={classes.card}>
      <CardContent>
        <HelpHeading />
        <HelpContent helpSteps={publishPackageSteps} />
      </CardContent>
      <CardActions>
        <Button color="primary" href="https://verdaccio.org/docs/en/installation" size="small" target="_blank">
          Learn More
        </Button>
      </CardActions>
    </Card>
  );
};

export default Help;
