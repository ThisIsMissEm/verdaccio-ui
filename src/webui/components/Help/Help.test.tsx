import React from 'react';
import { shallow } from 'enzyme';
import Help from '.';

describe('<Help /> component', () => {
  test('should render the component in default state', () => {
    const wrapper = shallow(<Help />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
