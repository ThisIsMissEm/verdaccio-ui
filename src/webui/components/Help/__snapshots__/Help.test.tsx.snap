// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Help /> component should render the component in default state 1`] = `"<div class=\\"MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 Hook-card-1atwj47\\"><div class=\\"MuiCardContent-root-29\\"><h2 class=\\"MuiTypography-root-30 MuiTypography-headline-35 MuiTypography-gutterBottom-57\\">No Package Published Yet.</h2><h3 class=\\"MuiTypography-root-30 MuiTypography-subheading-37 MuiTypography-colorTextSecondary-63 MuiTypography-gutterBottom-57 Hook-subheading-jtsb18\\">To publish your first package just:</h3><ul class=\\"MuiList-root-66 MuiList-padding-67\\"><li class=\\"MuiListItem-root-70 MuiListItem-default-73 MuiListItem-gutters-78 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38 Hook-interpunct-1jg88t5\\">1.</p><div class=\\"Hook-item-jwcf7z\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38\\">Login</p><div class=\\"css-1mta3t8 e9u09cy0\\"><span class=\\"css-1m8aenu e9u09cy1\\">npm adduser --registry http://localhost</span><button class=\\"MuiButtonBase-root-96 MuiIconButton-root-90 css-56v3u0 e9u09cy2\\" tabindex=\\"0\\" type=\\"button\\" title=\\"Copy to Clipboard\\"><span class=\\"MuiIconButton-label-95\\"><svg class=\\"MuiSvgIcon-root-99\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z\\"></path></svg></span></button></div></div></li><li class=\\"MuiListItem-root-70 MuiListItem-default-73 MuiListItem-gutters-78 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38 Hook-interpunct-1jg88t5\\">2.</p><div class=\\"Hook-item-jwcf7z\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38\\">Publish</p><div class=\\"css-1mta3t8 e9u09cy0\\"><span class=\\"css-1m8aenu e9u09cy1\\">npm publish --registry http://localhost</span><button class=\\"MuiButtonBase-root-96 MuiIconButton-root-90 css-56v3u0 e9u09cy2\\" tabindex=\\"0\\" type=\\"button\\" title=\\"Copy to Clipboard\\"><span class=\\"MuiIconButton-label-95\\"><svg class=\\"MuiSvgIcon-root-99\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z\\"></path></svg></span></button></div></div></li><li class=\\"MuiListItem-root-70 MuiListItem-default-73 MuiListItem-gutters-78 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38 Hook-interpunct-1jg88t5\\">3.</p><div class=\\"Hook-item-jwcf7z\\"><p class=\\"MuiTypography-root-30 MuiTypography-body2-38\\">Refresh this page</p></div></li></ul></div><div class=\\"MuiCardActions-root-108\\"><a class=\\"MuiButtonBase-root-96 MuiButton-root-111 MuiButton-text-113 MuiButton-textPrimary-114 MuiButton-flat-116 MuiButton-flatPrimary-117 MuiButton-sizeSmall-134 MuiCardActions-action-110\\" tabindex=\\"0\\" role=\\"button\\" href=\\"https://verdaccio.org/docs/en/installation\\" target=\\"_blank\\"><span class=\\"MuiButton-label-112\\">Learn More</span></a></div></div>"`;
